core/snippets/snippets_parser.py

```diff
@@ -191,18 +191,21 @@ def parse_file_content(file: str, text: str) -> list[SnippetDocument]:
 def parse_document(
     file: str, line: int, optional_body: bool, text: str
 ) -> Union[SnippetDocument, None]:
-    parts = re.split(r"^-\r?\n?$", text, flags=re.MULTILINE)
+    match = re.search(r"^-\r?\n?$", text, flags=re.MULTILINE)
```

Comment:
Think you can remove the `\r?\n?` from here as they will never match. Here's an example:

```python
>>> x = 'foo\n-\nbar\n-\nbaz'
>>> re.search(r'^-$', x, flags=re.MULTILINE)
<re.Match object; span=(4, 5), match='-'>
>>> re.search(r'^-\n$', x, flags=re.MULTILINE)
```

Reply:
Actually, even better, can just use `re.split(r"^-$", text, maxsplit=1, flags=re.MULTILINE)` — this will give you `context_text` and (optionally) `body_text` in a list.

Reply:
Done